### PR TITLE
Generate update query test

### PR DESCRIPTION
**Pull request by rushikesh-kulkarni**
Generate update query test

### CODE REVIEW

File: modules/crud/sql/update.go [hunk 1]
```diff
@@ -243,22 +292,66 @@ func flattenForDate(m *map[string]interface{}) error {
...
 	return nil
 }
+
+func (s *SQL) sanitiseUpdateQuery(sqlString string) string {
+	var placeholder byte
+	switch utils.DBType(s.dbType) {
+	case utils.Postgres, utils.SqlServer:
+		placeholder = '$'
+	}
+	log.Println("sqlString", sqlString, "PlaceHolder", placeholder)
```
Comment (sharadregoti) [inline modules/crud/sql/update.go]: is this log required

Comment (sharadregoti) [reply modules/crud/sql/update.go]: remove it

File: modules/crud/sql/update.go [hunk 2]
```diff
@@ -243,22 +292,66 @@ func flattenForDate(m *map[string]interface{}) error {
...
+				return fmt.Errorf("invalid current date value (%s) provided", val)
 			}
 		}
 	}
 	return nil
 }
+
+func (s *SQL) sanitiseUpdateQuery(sqlString string) string {
+	var placeholder byte
+	switch utils.DBType(s.dbType) {
```
Comment (sharadregoti) [inline modules/crud/sql/update.go]: use if statement here

File: modules/crud/sql/update.go [hunk 3]
```diff
@@ -243,22 +292,66 @@ func flattenForDate(m *map[string]interface{}) error {
...
+	case utils.Postgres, utils.SqlServer:
+		placeholder = '$'
+	}
+	log.Println("sqlString", sqlString, "PlaceHolder", placeholder)
+	var start bool
+	for i := 0; i < len(sqlString); i++ {
+		c := sqlString[i]
+		log.Println("c", c, placeholder)
```
Comment (sharadregoti) [inline modules/crud/sql/update.go]: remove log

File: modules/crud/sql/update.go [hunk 4]
```diff
@@ -243,22 +292,66 @@ func flattenForDate(m *map[string]interface{}) error {
...
+		}
+		if strings.HasPrefix(sqlString[i:], "WHERE") {
+			break
+		}
+	}
+	return sqlString
+}
+func (s *SQL) sanitiseUpdateQuery2(sqlString string) string {
+
+	for i := 0; i < len(sqlString); i++ {
```
Comment (sharadregoti) [inline modules/crud/sql/update.go]: u can use range statement here

File: modules/crud/sql/update_test.go [hunk 5]
```diff
@@ -2,49 +2,1366 @@ package sql
...
-		{name: "Successfull Test NOT IN", tableName: "fooTable", wantThis: "UPDATE fooTable SET String1=? WHERE (key1 NOT IN (?))", req: model.UpdateRequest{Update: map[string]interface{}{"$set": map[string]interface{}{"String1": "1"}}, Find: map[string]interface{}{"key1": map[string]interface{}{"$nin": 1}}}},
-
-		{name: "Error Update is NIL", tableName: "fooTable", wantThis: "UPDATE fooTable SET String1=? WHERE ((FindString1 = ?) AND (FindString2 = ?))", req: model.UpdateRequest{Find: map[string]interface{}{"FindString1": "1", "FindString2": "2"}}},
-		{name: "Error No $set", tableName: "fooTable", wantThis: "UPDATE fooTable SET String1=? WHERE ((FindString1 = ?) AND (FindString2 = ?))", req: model.UpdateRequest{Update: map[string]interface{}{}, Find: map[string]interface{}{"FindString1": "1", "FindString2": "2"}}},
-		{name: "Query Update sql", tableName: "fooTable", wantThis: "UPDATE fooTable SET String1=? WHERE ((FindString1 = ?) AND (FindString2 = ?))", req: model.UpdateRequest{Update: map[string]interface{}{"$set": map[string]interface{}{}}, Find: map[string]interface{}{"FindString1": "1", "FindString2": "2"}}},
+		{
+			name:   "test1",
```
Comment (sharadregoti) [inline modules/crud/sql/update_test.go]: provide a proper name for your test cases

**Comment by sharadregoti** [comment]
provide comments wherever necessary

File: modules/crud/sql/helpers.go [hunk 6]
```diff
@@ -73,15 +73,14 @@ func (s *SQL) generator(find map[string]interface{}) (goqu.Expression, []string)
...
+func (s *SQL) generateWhereClause(q *goqu.SelectDataset, find map[string]interface{}) (query *goqu.SelectDataset, arr []string) {
 	query = q
-	err = nil
 	if len(find) == 0 {
 		return
 	}
 	exp, arr := s.generator(find)
 	query = query.Where(exp)
-	return query, arr, err
+	return query, arr
```
[the remark this answers is not in the document]
Comment (rushikesh-kulkarni) [reply modules/crud/sql/helpers.go]: letting this be as discussed with Noorain

File: modules/crud/sql/update.go (hunk 4 above)
Comment (rushikesh-kulkarni) [reply modules/crud/sql/update.go]: Ranging causes iteration over `rune` values which leads to incorrect indexes. We need to iterate over the individual bytes in this case. Discussed with Noorain

**Pull request merged**